pkg/registry/extensions/daemonset/strategy.go

```diff
@@ -70,6 +70,14 @@ func (daemonSetStrategy) PrepareForUpdate(ctx genericapirequest.Context, obj, ol
 	// update is not allowed to set TemplateGeneration
 	newDaemonSet.Spec.TemplateGeneration = oldDaemonSet.Spec.TemplateGeneration
 
+	// update is not allowed to set Spec.Selector for v1beta2
+	// we assume it is not v1beta2 if no API version is found in RequestInfo
+	// TODO(#50791): after v1beta1 is retired, move selector immutability check to validation codes
+	requestInfo, found := genericapirequest.RequestInfoFrom(ctx)
+	if found && requestInfo.APIVersion == "v1beta2" {
```

Comment:
Refer this [comment](https://github.com/kubernetes/kubernetes/pull/50719/files#r133611825) for why checking API group may be unnecessary.